### ISSUE TITLE
Create a docker image with the contracts deployed on a private chain. 

### ISSUE THREAD

**Issue by CosminNechifor** [raiden-network/raiden-contracts#495]
# Feature request

We should provide an easier way for new users to interact with ``raiden-contracts`` and also ``Raiden`` on their private chain. @kelsos  was thinking about having this for testing it locally, in order to reduce overhead on setting up the ``private chain``.  

### Motivation:

- It would be useful for testing locally.
- It would allow new users and developers to install and run their ``private chain`` and interact with ``Raiden``. 
- Also if we set the ``ENTRYPOINT`` to ``bash`` the user will be able to do whatever they want with the containerized ``private chain`` (things such us install tmux, create new accounts and start new Raiden nodes).
- They would be able to even use for example: ``docker network create <network_name>`` to create a private network, which they can later use by providing the parameter ``--net`` when running the docker container ``docker run --rm -it -p 8545:8545 --net=<network_name> <name_of_builded_node>``
- (Not sure about) if we expose the right ports he could even connect to other users, which might be really useful again for testing/presentations.

I will be waiting on feedback. And if you agree with the proposal I would like to start working on this.

**Comment by hackaugusto** [comment]
How do you plan to implement this `docker image with the contracts deployed on a private chain`?

**Comment by pirapira** [comment]
A positive side effect: the docker script will serve as up-to-date documentation on how to deploy the contracts on a private chain.

**Comment by CosminNechifor** [comment]
Thank you for the question @hackaugusto . But I can not give a proper answer, because when I opened this I thought that I could create a single image that will start from a certain state (the private chain with contracts already deployed), I am still looking if this is possible. 

Another approach I arrived at, after looking into the first one is using  ``docker-compose`` in order to do the following things:

1. Instantiate a container that is running parity and persistent data directory to  ``~/.local/share/io.parity.ethereum/<private_chain_name>``.
2. After parity starts, will deploy the contracts using the scripts we have in this repository.
3. Final step will be to run Raiden also in another container. 

Will be looking more into that possibility.

**Comment by karlb** [comment]
Is there still interest in this, or should we close the issue as "won't do"?

**Comment by kelsos** [comment]
I would say you can close it, and point people to the integration image of light client if they need a private chain docker image. https://hub.docker.com/r/raidennetwork/lightclient-integration (it also has pfs and two python nodes as well as a functioning private chain)
